server/models/migrations/next/migrate.sql

```diff
@@ -275,4 +275,89 @@ INSERT INTO `unit` VALUES
 INSERT INTO `unit` VALUES 
 (238, 'Indicators report', 'TREE.INDICATORS_REPORT', 'Rapport sur les indicateurs', 144,'/modules/reports/indicatorsReport', '/reports/indicatorsReport');
 INSERT INTO `report` (`id`, `report_key`, `title_key`) 
-VALUES  (31, 'indicatorsReport', 'TREE.INDICATORS_REPORT');
\ No newline at end of file
+VALUES  (31, 'indicatorsReport', 'TREE.INDICATORS_REPORT');
+
+
+
+
+-- jeremie lodi
+DELIMITER $$
+
+-- this is handled on change rate changed
+DROP PROCEDURE IF EXISTS `UpdateLossAccount`$$
+CREATE   PROCEDURE `UpdateLossAccount`(
+  IN _enterpriseId INT,
+  IN _project_id INT, 
+  IN _user_id INT,
+  IN _currencyId INT,
+  IN _date DATE, 
+  IN _rate DECIMAL(19, 4)
+  )
+BEGIN
+DECLARE _account_id, _loss_account_id, _gain_account_id INT(10) UNSIGNED;
+DECLARE _privous_rate, _balance, _difference DECIMAL(19, 4);
+DECLARE _fiscalyear_id, _period_id  mediumint(8) UNSIGNED;
+DECLARE voucherUUID BINARY(16);
+DECLARE _description, _account_label VARCHAR(255);
+
+DECLARE done BOOLEAN;
+DECLARE curs1 CURSOR FOR 
+  SELECT c.account_id , CONCAT(a.number , ' - ', a.label) 
+  FROM cash_box_account_currency c
+  JOIN account a ON a.id =  c.account_id
+  WHERE currency_id = _currencyId;
+   
+DECLARE CONTINUE HANDLER FOR NOT FOUND SET done = TRUE;
+
+SET _privous_rate = (SELECT GetExchangeRate(_enterpriseId, _currencyId, _date)  LIMIT 1);
+
+SET _fiscalyear_id = (SELECT f.id FROM fiscal_year f WHERE f.start_date <= DATE(_date) AND DATE(_date) <= f.end_date  LIMIT 1);
+
+SET _period_id = (SELECT p.id FROM period p WHERE p.start_date <= DATE(_date) AND DATE(_date) <= p.end_date LIMIT 1);
+
+SET _loss_account_id = (SELECT loss_account_id FROM enterprise WHERE id = _enterpriseId);
+SET _gain_account_id = (SELECT gain_account_id FROM enterprise Where id= _enterpriseId);
+
+SET _description = "Gain ou perte pour le nouveau taux d'echange(Gain Or Loss for New exchange rate)";
+
+OPEN curs1;
+    read_loop: LOOP
+    
+    FETCH curs1 INTO _account_id, _account_label;
+    
+    IF done THEN
+      LEAVE read_loop;
+    END IF;
+        
+    SET _balance = ( 
+		 	SELECT SUM(debit - credit) 
+			FROM period_total 
+			WHERE account_id = _account_id and fiscal_year_id= _fiscalyear_id
+      GROUP BY fiscal_year_id
+      );
```

Comment:
Maybe it's just Github, but this formatting is pretty hard to read.  Please make sure you format your code with two spaced indents for each code block so that it is easier to review.

Reply:
it's Github